lib/index.js

```diff
@@ -108,15 +108,12 @@ api.set = async ({
     // FIXME: may need to allow up to N tokens, one per `clientId` if
     // `clientId` is set
 
-    // if there is an existing, unexpired nonce, reuse it, do not generate
-    // a new one to avoid DoS attack of constantly running bcrypt algorithm
+    // if there is an existing nonce, delete it, and generate a new one
     try {
-      const existingToken = await api.get({actor: null, account, email, type});
-      // keep existing token if it hasn't expired; note that there is no
-      // `challenge` to return because it is not known in the database
-      if(existingToken && new Date() < new Date(existingToken.expires)) {
-        return {type};
-      }
+      // check to see if nonce exists
+      await api.get({actor: null, account, email, type});
+      // the `get` API call did not throw, a nonce exists, delete it
+      await api.remove({actor, account, type});
```

Comment:
I see that the `remove` API can also throws a `NotFoundError`.  In this case, if that were to happen, that error would be thrown creating confusion because we would not know which API call threw.

The only way this can happen however, is if there was a race where another process removed the token between L114 and L115.  So, in that instance, a `NotFoundError` from the `remove` API should just be discarded but any other error should be thrown.  The opposite behavior from the conditional on L118.  So this should be refactored to take all this into account.

Reply:
Meant to include link: https://github.com/digitalbazaar/bedrock-authn-token/blob/127bddfa2eb8b99b2403a500ec181ff5cfaa2c37/lib/index.js#L299